pkg/client/client.go

```diff
@@ -30,16 +31,18 @@ type ImageClient interface {
 // Client is a container image registry client to list tags of given image
 // URLs.
 type Client struct {
-	gcr    *gcr.Client
-	docker *docker.Client
-	quay   *quay.Client
+	gcr        *gcr.Client
+	docker     *docker.Client
+	quay       *quay.Client
+	selfhosted *selfhosted.Client
```

Comment:
We can follow up with another PR to have multiple selfhosted registries for different hosts.

These can then become a slice of `ImageClient`.